src/torchmetrics/metric.py

```diff
@@ -860,68 +861,72 @@ def __hash__(self) -> int:
 
         return hash(tuple(hash_vals))
 
-    def __add__(self, other: "Metric") -> "CompositionalMetric":
+    def __add__(self, other: Union["Metric", int, builtins.float, Tensor]) -> "CompositionalMetric":
```

Comment:
```suggestion
    def __add__(self, other: Union["Metric", int, float, Tensor]) -> "CompositionalMetric":
```

why not just `float`?

Reply:
Because `Metric` has a function called `float`. This function was used instead of the built-in `float` type.

Reply:
let's add this note when you import `builtins`